src/contributors/MapContributor.ts

```diff
@@ -2083,6 +2096,17 @@ export class MapContributor extends Contributor {
                 });
                 aggregation.fetch_hits.include = Array.from(fetchSet);
             }
+            if (ls.includeFields && ls.includeFields.size > 0) {
+                if (!aggregation.fetch_hits) {
+                    aggregation.fetch_hits = {size: 1};
+                    aggregation.fetch_hits.include = [];
+                }
+                const fetchSet = new Set<string>(aggregation.fetch_hits.include);
```

Comment:
`new Set(...aggregation.fetch_hits.include, ...ls.includeFields)` may do the same